clustering/singleton/extension/src/main/java/org/wildfly/extension/clustering/singleton/SingletonPolicyResourceDefinition.java

```diff
@@ -68,26 +71,44 @@ static PathElement pathElement(String value) {
     }
 
     enum Attribute implements org.jboss.as.clustering.controller.Attribute {
-        CACHE_CONTAINER("cache-container", ModelType.STRING, null),
-        CACHE("cache", ModelType.STRING, new ModelNode(SubGroupServiceNameFactory.DEFAULT_SUB_GROUP)),
+        CACHE_CONTAINER("cache-container", ModelType.STRING, false),
+        CACHE("cache", ModelType.STRING, true),
         QUORUM("quorum", ModelType.INT, new ModelNode(1)),
         ;
         private final SimpleAttributeDefinition definition;
 
+        private Attribute(String name, ModelType type, boolean allowNull) {
+            this.definition = createBuilder(name, type).setAllowNull(allowNull).build();
+        }
+
         private Attribute(String name, ModelType type, ModelNode defaultValue) {
-            this.definition = new SimpleAttributeDefinitionBuilder(name, type)
+            this.definition = createBuilder(name, type)
                     .setAllowExpression(true)
```

Comment:
Yes - but this will be addressed in phase 3 (see the PR description), including:
- capability references for these attributes
- disallow expressions (because these are model references) + appropriate rejection transformation